syntax = "proto3";

// Defines RPC for the Trainer to call on the Pokemon
package pokemon;

// Message to begin training
message InitDeviceRequest {
    bytes data = 1;
    int32 layers = 2;
}

message InitDeviceResponse {
    bool success = 1;
    string message = 2;
}

// Message to begin training
message StartTrainingRequest {}

message StartTrainingResponse {
    string message = 1; // Success
    bool success = 2;
}

// Service definition for Trainer
service PokemonService {
    rpc InitDevice(InitDeviceRequest) returns (InitDeviceResponse);
    rpc StartTraining(StartTrainingRequest) returns (StartTrainingResponse);
}